site/docs/tutorial/sql_to_substrait.md

```diff
@@ -0,0 +1,995 @@
+# SQL to Substrait tutorial
+
+This is an introductory tutorial to learn the basics of Substrait for readers
+already familiar with SQL. We will look at how to construct a Substrait plan
+from an example query.
+
+We'll present the Substrait in JSON form to make it relatively readable to
+newcomers. Typically Substrait is exchanged as a protobuf message, but for
+debugging purposes it is often helpful to look at a serialized form. Plus,
+it's not uncommon for unit tests to represent plans as JSON strings. So
+if you are developing with Substrait, it's useful to have experience reading
+them.
+
+!!! note
+    Substrait is currently only defined with Protobuf. The JSON provided here is the
+    Protobuf JSON output, but it is *not* the official Substrait text format. Eventually,
+    Substrait will define it's own human-readable text format, but for now this
+    tutorial will make due with what Protobuf provides.
+
+Substrait is designed to communicate plans (mostly logical plans). Those plans
+contain types, schemas, expressions, extensions, and relations. We'll look at 
+them in that order, going from simplest to most complex until we can construct
+full plans.
+
+![](substrait_components.svg)
+
+This tutorial won't cover all the details of each piece, but it will give you an
+idea of how they connect together. For a detailed reference of each individual 
+field, the best place to look is reading the 
+[protobuf definitions](https://github.com/substrait-io/substrait/tree/main/proto/substrait).
+They represent the source-of-truth of the spec and are well-commented to address
+ambiguities.
+
+## Problem Set up
+
+To learn Substrait, we'll build up to a specific query. We'll be using the
+tables:
+
+```sql
+CREATE TABLE orders (
+  product_id: i64 NOT NULL,
+  quantity: i32 NOT NULL,
+  order_date: date NOT NULL,
+  price: decimal(10, 2)
+);
+```
+
+```sql
+CREATE TABLE products (
+  product_id: i64 NOT NULL,
+  categories: list<string NOT NULL> NOT NULL,
+  details: struct<manufacturer: string, year_created: int32>,
+  product_name: string
+);
+```
+
+This `orders` table represents events where products were sold, recording how
+many (`quantity`) and at what price (`price`). The `products` table provides
+details for each product, with `product_id` as the primary key.
+
+And we'll try to create the query:
+
+```sql
+SELECT
+  product_name,
+  product_id,
+  sum(quantity * price) as sales
+FROM
+  orders
+INNER JOIN
+  products
+ON
+  orders.product_id = products.product_id
+WHERE
+  -- categories does not contain "Computers"
+  INDEX_IN("Computers", categories) IS NULL
+GROUP BY
+  product_name,
+  product_id
+```
+
+The query asked the question: For products that aren't in the `"Computer"`
+category, how much has each product generated in sales?
+
+However, Substrait doesn't correspond to SQL as much as it does to logical plans.
```

Comment:
```suggestion
However, Substrait doesn't correspond to SQL as much as it does to query plans.
```

Minor nit: Substrait could be used to represent both logical and physical plans.  This sentence _could_ potentially lead someone to thinking it is only for logical plans.